src/Misc/layoutbin/powershell/Start-AzpTask.ps1

```diff
@@ -0,0 +1,146 @@
+<# 
+A PowerShell script that is used to invoke a VSTS task script. This script is used by the VSTS task runner to invoke the task script.
+This script replaces some legacy stuff in PowerShell3Handler.cs and turns it into a dedicated signed script. 
+since it is parameterized it can be signed and trusted for WDAC and CLM.
+This is the original code that has been repalced - 
+@". ([scriptblock]::Create('if ([Console]::InputEncoding -is [Text.UTF8Encoding] -and [Console]::InputEncoding.GetPreamble().Length -ne 0) {{ [Console]::InputEncoding = New-Object Text.UTF8Encoding $false }} if (!$PSHOME) {{ $null = Get-Item -LiteralPath ''variable:PSHOME'' }} else {{ Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'')) ; Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1'')) }}')) 2>&1 | ForEach-Object {{ Write-Verbose $_.Exception.Message -Verbose }} ; Import-Module -Name '{0}' -ArgumentList @{{ NonInteractive = $true }} -ErrorAction Stop ; $VerbosePreference = '{1}' ; $DebugPreference = '{1}' ; Invoke-VstsTaskScript -ScriptBlock ([scriptblock]::Create('. ''{2}'''))",
+#>
+
+param ( 
+    [Parameter(mandatory = $true)]$name, # Argument 0 
+    [Parameter(mandatory = $true)]$DebugOption, # Argument 1
+    [Parameter(mandatory = $true)]$scriptBlockString	# Argument 2 
+)
+
+function GetCLM {
+    # This is new functionality to detect if we are running in a constrained language mode.
+    # This is only used to display debug data if the device is in CLM mode by default.
+    # This test is modeled after the traditional __PSSCRIPTPOLICYTEST_LQU1DAME file
+    # This creates a temporary file with a changing signature 
+    # which will return the execution context language mode. 
+    $tempFileGuid = New-Guid | Select-Object -Expand Guid 
+    $tempFile = "$($env:temp)\$($tempFileGuid).ps1"
+    # Use the command which not allowed in constrained language mode.
+    Write-Output 'New-Object -TypeName System.Collections.ArrayList' | Out-File -FilePath $tempFile -append 
+    # Now we execute the file and return the result. We need to try catch this because of how PS handles 
+    # the execution of CLM files in a non CLM environment.
+
+    try {
+        . $tempFile
+        $status = "FullLanguage"
+    }
+    catch [System.Management.Automation.PSNotSupportedException] {
+        $status = "ConstrainedLanguage"
+    }
+
+    Remove-Item $tempFile 
+    return $status 
+}
+
+# This is the notes we want to display if CLM is active on the device by default.
+$verboseNotes = @"
```

Comment:
I think the most part of these notes can be moved to a separate .md doc and only the url can be given here